core/src/mindustry/io/TypeIO.java

```diff
@@ -365,6 +365,19 @@ public static Rules readRules(Reads read){
         return JsonIO.read(Rules.class, string);
     }
 
+    public static void writeAmendments(Writes write, Amendments amendments){
+        String string = JsonIO.write(amendments);
```

Comment:
Writing large amounts of data with JSON is risky; with the 32/64k character limit, you could easily run into IO issues.

Reply:
so i would have to send it in parts similar to block snapshots or something? (assuming the limit is packet size and not your jsonio thing)